x-pack/plugin/monitoring/build.gradle

```diff
@@ -28,13 +28,13 @@ configurations {
   testArtifacts.extendsFrom testRuntime
   testArtifacts.extendsFrom testImplementation
 }
-task testJar(type: Jar) {
+def testJar = tasks.register("testJar", Jar) {
   appendix 'test'
   from sourceSets.test.output
 }
 artifacts {
   // normal es plugins do not publish the jar but we need to since users need it for extensions
-  archives jar
+  archives tasks.named("jar")
   testArtifacts testJar
```

Comment:
Same as above, shouldn't we use a task provider for `testJar` as well?

Reply:
it is the task provider. I made the variable typed so its more clear